pkg/utils/kubernetes/secretref.go

```diff
@@ -32,26 +31,6 @@ func GetSecretByReference(ctx context.Context, c client.Reader, ref *corev1.Secr
 	return secret, nil
 }
 
-// CreateOrUpdateSecretByReference creates or updates the secret referenced by the given secret reference
-// with the given type, data, and owner references.
-func CreateOrUpdateSecretByReference(ctx context.Context, c client.Client, ref *corev1.SecretReference, secretType corev1.SecretType, data map[string][]byte, ownerRefs []metav1.OwnerReference) error {
```

Comment:
Originally, I wanted to do exactly this, but this would have meant to call `pkg/controllerutils.CreateOrStrategicMergePatch` from the `pkg/utils/kubernetes` package. This introduced a cyclic dependency as `pkg/controllerutils` is already depending on `pkg/utils/kubernetes`, so I decided to simply remove the function for now as the additional value compared to calling `pkg/controllerutils.CreateOrStrategicMergePatch` was very little.

Reply:
OK, thanks for the explanation.